cfgov/agreements/metadata.py

```diff
@@ -0,0 +1,32 @@
+from django.conf import settings
+
+import boto3
+
+
+class Metadata:
+    def __init__(self):
+        s3 = boto3.resource("s3")
+        self.bucket = s3.Bucket(settings.AWS_STORAGE_BUCKET_NAME or "")
+        self.prefix = "a/assets/bulk_agreements/"
+        self.flexibilities = [
+            "Q1-2020",
+            "Q2-2020",
+            "Q3-2020",
+            "Q4-2020",
+            "Q1-2021",
+        ]
+        self.notes = {
+            "Q2-2019": "Agreements are incomplete due to technical submission"
+            " issues at the Bureau"
+        }
+
+    def get_objects_by_prefix(self):
+        return self.bucket.objects.filter(Prefix=self.prefix)
+
+    def get_sorted_agreements(self, reverse=True):
+        try:
+            agreements = [f.key for f in self.get_objects_by_prefix()]
+            agreements.sort(reverse=reverse)
+            return agreements
+        except Exception:
```

Comment:
What exception are we expecting to see here, just a failure to connect to S3? It's generally better practice in Python to catch a specific exception rather than a generic one. 

Honestly, if some unexpected kind of exception happens here, I'd prefer a real error to bubble up, so it can get logged and we can investigate it. That would resolve my question from my original comment ("Is there also a way to notify cf.gov devs if this is broken?"). But we could ask real cf.gov back-end devs what they consider best practice to be.

Reply:
The error now bubbles in 35bb8885e8bd8034abff57dbb3705d959f5be48a